src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1820,20 +1820,32 @@ bool checkValid(MethodSymbol candidate, bool isCandidate, DiagnosticBag specific
                         diagnostics.Add(ErrorCode.ERR_NoMainInClass, mainType.Locations.First(), mainType);
                     }
                 }
-                else if (viableEntryPoints.Count > 1)
-                {
-                    viableEntryPoints.Sort(LexicalOrderSymbolComparer.Instance);
-                    var info = new CSDiagnosticInfo(
-                         ErrorCode.ERR_MultipleEntryPoints,
-                         args: Array.Empty<object>(),
-                         symbols: viableEntryPoints.OfType<Symbol>().AsImmutable(),
-                         additionalLocations: viableEntryPoints.Select(m => m.Locations.First()).OfType<Location>().AsImmutable());
-
-                    diagnostics.Add(new CSDiagnostic(info, viableEntryPoints.First().Locations.First()));
-                }
                 else
                 {
-                    entryPoint = viableEntryPoints[0];
+                    foreach (var viableEntryPoint in viableEntryPoints)
+                    {
+                        if (viableEntryPoint.UnmanagedCallersOnlyAttributeData is { } data)
+                        {
+                            Debug.Assert(!ReferenceEquals(data, UnmanagedCallersOnlyAttributeData.Uninitialized));
+                            diagnostics.Add(ErrorCode.ERR_EntryPointCannotBeUnmanagedCallersOnly, viableEntryPoint.Locations.First());
```

Comment:
>ERR_EntryPointCannotBeUnmanagedCallersOnly [](start = 54, length = 42)

Why can't the entry point be marked `[UnmanagedCallersOnly]`?

Reply:
A method marked with `UnmanagedCallersOnly` cannot be directly called from managed code. `Main` is no exception.

---
In reply to: [482326757](https://github.com/dotnet/roslyn/pull/47311#discussion_r482326757) [](ancestors = 482326757)

Reply:
Then won't we report an error at the callsite if this entry point is called from managed code?

---
In reply to: [482329176](https://github.com/dotnet/roslyn/pull/47311#discussion_r482329176) [](ancestors = 482329176,482326757)

Reply:
It's an entry point. There is no callsite in user code, but CLR will still call it and expect it to be a managed method. Same with module initializers.

---
In reply to: [482336042](https://github.com/dotnet/roslyn/pull/47311#discussion_r482336042) [](ancestors = 482336042,482329176,482326757)